docs/howto/danceable-lyrics-input-tool/README.md

```diff
@@ -0,0 +1,561 @@
+
+We use the [Polars](https://www.pola.rs/) for data processing. Polars has many benefits over the more common Pandas, not limited to:
+* Extremely **fast**
+* Polars can optimize queries and perfrom parallel processing for free
+* A very familiar API to expressively manipulate `DataFrame`'s
+* Zero-copy support with [Apache Arrow](https://arrow.apache.org/), our native SDK format
+
+> :information_source: For more information on Polars, see [the User Guide](https://pola-rs.github.io/polars-book/user-guide/) and the [Python API](https://pola-rs.github.io/polars/py-polars/html/reference/).
+
+## Table of Contents
+* [Download Input Data](#download-input-data)
+* [Basic Plugin Setup](#basic-plugin-setup)
+* [Write a Plugin](#write-a-plugin)
+    * [Dependencies](#1-dependencies)
+    * [Imports](#2-imports)
+    * [Initialization](#3-initialization)
+    * [Data Processing](#4-data-processing)
+    * [Putting It All Together](#5-putting-it-all-together)
+* [Packaging into a YXI](#package-into-a-yxi)
+* [Run the Test Client](#run-the-test-client)
+* [Install in Designer](#install-and-run-in-designer)
+    * [Method 1](#method-1)
+    * [Method 2](#method-2)
+* [Run in Designer](#run-in-designer)
+
+## Download Input Data
+Download and extract the required dataset files. You will reference them later in this guide. We provide 2 datasets—the full dataset and a smaller, truncated version.
+
+* The full version expands to around 11 GB of data. You can [download it here](https://drop.alteryx.com/public/file/SAlShV5VqkOI5_Jyx1me3w/Danceable_Lyrics_demo_data-Inspire-2023.7z).
+* The truncated version expands to around 5.7 GB of data. You can [download it here](https://drop.alteryx.com/public/file/yHv77vLzKkCXHj8-OWVg0Q/Danceable_Lyrics_demo_data-Inspire-(Truncated)-2023.7z).
+
+Go to the `README.md` files within the respective archives for source information.
+
+Before proceeding, let's take a quick look at the data set files:
+![head -n 170 .\genius_song_lyrics.csv | rich --csv -](./assets/genius-lyrics-head.png)
+
+Above, we see `genius_song_lyrics.csv` is made up of the columns `title`, `tag`, `artist`, `year`, `views`, `features`, `lyrics`, `id`, `language_cld3`, `language_ft`, and `language`. Notice that `lyrics` contains strings with multiple lines. Let's look at this just a bit closer:
+
+![head -n 5 .\genius_song_lyrics.csv](./assets/genius-lyrics-head2.png)
+
+We can see the strings are quoted. This is great, Polars supports all of this right out of the box!
+
+Moving on to our input files containing track features:
+![tail --bytes +4 ./artists.csv | rich -h 5 --csv -](./assets/artists-head.png)
+
+Above, we pipe commands through `tail --bytes +4` in order to skip past the first 4 bytes representing a UTF-8 [Byte Order Mark (BOM)](https://en.wikipedia.org/wiki/Byte_order_mark) in our input CSV files. But this is something Polars can handle for us. Let's try another approach using Python and Polars:
```

Comment:
TW review:

Above, we pipe commands through `tail --bytes +4` in order to skip past the first 4 bytes, which represent a UTF-8 [Byte Order Mark (BOM)](https://en.wikipedia.org/wiki/Byte_order_mark) in our input CSV files. But this is something Polars can handle for us. Let's try another approach using Python and Polars: